### PR TITLE
User Assigned Identity: add corresponding data source

### PR DESCRIPTION
**Pull request by logachev**
- Created data source for User Assigned Identity
- Added relevant tests
- Updated documentation

### CODE REVIEW

File: website/docs/d/user_assigned_identity.html.markdown [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+output "uai_principal_id" {
+  value = "${data.azurerm_user_assigned_identity.test.principal_id}"
+}
+```
+
+## Argument Reference
+
+* `name` - (Required) Specifies the name of the user assigned identity.
+* `resource_group_name` - (Required) Specifies the name of the resource group.
```
Comment (katbyte) [inline website/docs/d/user_assigned_identity.html.markdown]: Could we use wording consistant with the rest of the provider:
```suggestion
* `resource_group_name` - (Required) The Name of the Resource Group in which the User Assigned Identity exists.
```

File: website/docs/d/user_assigned_identity.html.markdown [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+## Attributes Reference
```
Comment (katbyte) [inline website/docs/d/user_assigned_identity.html.markdown]: I think you missed adding location to the documentation?

File: website/docs/d/user_assigned_identity.html.markdown (hunk 1 above)
Comment (logachev) [reply website/docs/d/user_assigned_identity.html.markdown]: It depends which resource you use as a reference :) (I think I was looking at virtual machine & virtual network)

File: azurerm/data_source_user_assigned_identity_test.go [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+					resource.TestCheckResourceAttr(dataSourceName, "tags.%", "1"),
+					testEqualResourceAttr(dataSourceName, resourceName, "principal_id"),
+					testEqualResourceAttr(dataSourceName, resourceName, "client_id"),
+				),
+			},
+		},
+	})
+}
+
+func testEqualResourceAttr(dataSourceName string, resourceName string, attrName string) resource.TestCheckFunc {
```
Comment (logachev) [inline azurerm/data_source_user_assigned_identity_test.go]: I noticed that in other data source tests there is no any checks that data source resource contains same information as a resource.
Do you think this is valuable to move this function to some shared place and reuse in other data source tests?

**Pull request merged**